File: frontend/src/types/fx.ts
/** Matches the existing FastAPI /fx/usdcop/* response contracts. */
export interface FxDailyPoint {
  trade_date: string;
  open_price: number;
  high_price: number;
  low_price: number;
  close_price: number;
  change_1d_cop: number | null;
  change_1d_pct: number | null;
}

export interface FxHistoryResponse {
  pair: string;
  source: string;
  latest_date: string;
  points: FxDailyPoint[];
}

File: backend/src/colombia_markets/schemas/fx.py
"""Public USD/COP response schemas. All prices are COP per one USD."""

from datetime import date

from pydantic import BaseModel, Field


class FxDailyPoint(BaseModel):
    trade_date: date
    open_price: float
    high_price: float
    low_price: float
    close_price: float
    change_1d_cop: float | None = None
    change_1d_pct: float | None = None


class FxHistoryResponse(BaseModel):
    pair: str = "USD/COP"
    source: str = "yahoo_finance"
    latest_date: date
    points: list[FxDailyPoint] = Field(default_factory=list)


class FxLatestResponse(BaseModel):
    pair: str = "USD/COP"
    source: str = "yahoo_finance"
    latest_date: date
    point: FxDailyPoint
